### PR TITLE
[rllib] Fix Multidiscrete support

### LINKED ISSUE #4866: [rllib] MultiDiscrete action distributions aren't handled correctly in all algorithms
### Describe the problem
Currently only IMPALA and APPO handle MultiDiscrete properly. We should fix this or raise an error telling users to use Tuple instead.

As a workaround, you can use Tuple[Discrete, Discrete, ...] instead of MultiDiscrete.

### Source code / logs
This is from https://groups.google.com/forum/#!topic/ray-dev/RaqWO89Rl6E

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

Properly support MultiDiscrete so that we don't need to use Tuple[Discrete, ...]. Add it to the supported spaces test.

## Related issue number

Closes https://github.com/ray-project/ray/issues/4866

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

File: python/ray/rllib/agents/impala/vtrace_policy.py [hunk 1]
```diff
@@ -258,32 +256,13 @@ def make_time_major(tensor, drop_last=False):
...
-            dist_class=dist_class,
+            dist_class=Categorical if is_multidiscrete else dist_class,
             valid_mask=make_time_major(mask, drop_last=True),
             vf_loss_coeff=self.config["vf_loss_coeff"],
             entropy_coeff=self.config["entropy_coeff"],
             clip_rho_threshold=self.config["vtrace_clip_rho_threshold"],
             clip_pg_rho_threshold=self.config["vtrace_clip_pg_rho_threshold"])
 
-        # KL divergence between worker and learner logits for debugging
-        model_dist = MultiCategorical(unpacked_outputs)
```
Comment (ericl) [inline python/ray/rllib/agents/impala/vtrace_policy.py]: Removed the KL stats since I doubt they were very useful

Comment (stefanpantic) [reply python/ray/rllib/agents/impala/vtrace_policy.py]: I think they should stay in. They can be useful for debugging the policy (e.g. large fluctuations are a bad sign). See slide 21 [here](http://rll.berkeley.edu/deeprlcourse/docs/nuts-and-bolts.pdf).

**Comment by ericl** [comment]
Ping on review here. Let me know if don't have the time.

File: python/ray/rllib/agents/impala/vtrace_policy.py [hunk 2]
```diff
@@ -191,9 +191,7 @@ def __init__(self,
         unpacked_outputs = tf.split(
```
Comment (hartikainen) [inline python/ray/rllib/agents/impala/vtrace_policy.py]: Is this needed anymore?

File: python/ray/rllib/tests/test_supported_spaces.py [hunk 3]
```diff
@@ -17,6 +17,7 @@
 ACTION_SPACES_TO_TEST = {
     "discrete": Discrete(5),
     "vector": Box(-1.0, 1.0, (5, ), dtype=np.float32),
+    "multidiscrete": MultiDiscrete([1, 2, 3, 4]),
```
Comment (hartikainen) [inline python/ray/rllib/tests/test_supported_spaces.py]: Should there be an explicit test for the case that failed earlier? It's possible that this test already covers it though.

**Review by hartikainen — APPROVED** [review]
Apart from the couple of minor comments, this looks good to me.

File: python/ray/rllib/agents/impala/vtrace_policy.py (hunk 2 above)
Comment (ericl) [reply python/ray/rllib/agents/impala/vtrace_policy.py]: It's used below in a spot.

File: python/ray/rllib/tests/test_supported_spaces.py (hunk 3 above)
Comment (ericl) [reply python/ray/rllib/tests/test_supported_spaces.py]: Yep, this covers the failing case. The test fails before the other changes in the PR.

**Pull request merged**